hail/python/hailtop/batch/job.py

```diff
@@ -1045,60 +1043,25 @@ def prepare_argument_for_serialization(arg):
                                       for name, resource in arg._resources.items()})
             return ('value', arg)
 
-        def deserialize_argument(arg):
-            typ, val = arg
-            if typ == 'py_path':
-                return dill.load(open(val, 'rb'))
-            if typ in ('path', 'dict_path'):
-                return val
-            assert typ == 'value'
-            return val
-
-        def wrap(f):
-            @functools.wraps(f)
-            def wrapped(*args, **kwargs):
-                args = [deserialize_argument(arg) for arg in args]
-                kwargs = {kw: deserialize_argument(arg) for kw, arg in kwargs.items()}
-                return f(*args, **kwargs)
-            return wrapped
-
-        for i, (result, unapplied, args, kwargs) in enumerate(self._functions):
+        for i, (result, unapplied_id, args, kwargs) in enumerate(self._function_calls):
+            unapplied = self._batch._get_python_function(unapplied_id)
```

Comment:
I don't think this is used, can we delete this?

Reply:
It's still used to generate the `self._user_code` below, which I'm under the impression is used for debugging, but we can move the definition down to where it's actually used at least.